internal/cli/app/app_list_test.go

```diff
@@ -0,0 +1,170 @@
+package app
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func TestAppListSetup(t *testing.T) {
+	t.Run("Setup creates a realm client with a session", func(t *testing.T) {
+		config := cli.CommandConfig{RealmBaseURL: "http://localhost:8080"}
+		cmd := &appListCommand{}
+
+		authToken := "authToken"
+		refreshToken := "refreshToken"
+		profile, profileErr := cli.NewProfile(primitive.NewObjectID().Hex())
+		assert.Nil(t, profileErr)
+
+		profile.SetSession(authToken, refreshToken)
+
+		err := cmd.Setup(profile, nil, config)
+		assert.Nil(t, err)
+		assert.NotNil(t, &cmd.realmClient)
+	})
+}
+
+func TestAppListHandler(t *testing.T) {
+	project1 := "project1"
+	project2 := "project2"
+
+	appName1 := "name1"
```

Comment:
to help test code readability, can we try to omit the variables that do not need to be declared (in that they aren't used anywhere else downstream).  at a glance here, it looks like `appNameX` and all the `role` stuff can be inlined where its used

in general, I'd like to try to keep test code as direct and quick to read as possible.  every line counts :)